lib/index.js

```diff
@@ -22,11 +22,36 @@ var AppsFlyer = module.exports = integration('AppsFlyer')
  */
 
 AppsFlyer.ensure(function(msg, settings) {
-  if (contains(msg.library(), 'ios') && !settings.appleAppID) {
+  
+  // Check to ensure msg has a device contextual property
+  if (!msg.device()) {
+    return this.reject('Events to AppsFlyer must specify device type and advertisingId as contextual properties');
+  }
+  
+  // Check to ensure message has advertisingId as a property of device object
+  var deviceInfo = msg.device();
```

Comment:
Agree on moving it up to line 25. The fallback to an empty object might not be necessary though as the program will reject if msg.device returns undefined right?